### PR TITLE
protractorAdapter: Connect to test suit AND fix all broken methods

### PR DESCRIPTION
**Pull request by hepiyellow**
I suggest reviewing this PR commit-by-commit.
It's self-explanatory.

### Jasmine
- I could get Protractor to work with `mocha` and handle async test function, so it works with jasmine.
- I used `protractor.conf.ts` to start the server, instead of passing `before/after` methods to the test-suit

----------------
@mykas FYI

### CODE REVIEW

File: adapters/protractor/src/index.ts [hunk 1]
```diff
@@ -1,9 +1,20 @@
+import {browser, ElementFinder} from 'protractor';
+import {Key as SeleniumKey} from 'selenium-webdriver';
 import {Locator, UniDriverList, UniDriver, MapFn, waitFor, NoElementWithLocatorError, MultipleElementsWithLocatorError, isMultipleElementsWithLocatorError} from '@unidriver/core';
-import {browser, ElementFinder}                   from 'protractor';
-
 type ElementGetter = () => Promise<ElementFinder | null>;
 type ElementsGetter = () => Promise<ElementFinder[]>;
 
+const CamelCaseToHyphen = (key: string) => key.replace(/([a-z])([A-Z])/g, '$1_$2');
```
Comment (GabiGrin) [inline adapters/protractor/src/index.ts]:
```suggestion
const camelCaseToHyphen = (key: string) => key.replace(/([a-z])([A-Z])/g, '$1_$2');
```

File: test-suite/src/run-test-suite.ts [hunk 2]
```diff
@@ -128,6 +128,35 @@ export const runTestSuite = (params: TestSuiteParams) => {
 
         });
 
+        describe('hasClass()', () => {
```
Comment (GabiGrin) [inline test-suite/src/run-test-suite.ts]: nice!

File: adapters/selenium/src/index.ts [hunk 3]
```diff
@@ -130,7 +130,7 @@ export const seleniumUniDriver = (wep: WebElementGetter): UniDriver<WebElement>
     hasClass: async (className: string) => {
       const el = await elem();
       const cl = await el.getAttribute('class');
-      return cl.indexOf(className) !== -1;
+      return cl.split(' ').filter(c => c === className).length > 0;
```
Comment (GabiGrin) [inline adapters/selenium/src/index.ts]:
```suggestion
      return cl.split(' ').includes(className);
```

**Comment by hepiyellow** [comment]
@GabiGrin Yes it passes locally.

**Comment by hepiyellow** [comment]
@ranyitz @hadar111 

### `webdriver-manager update` params
I set up a protractor test env for Unidriver protractor adapter (Not using Yoshi)

#### With Yoshi config
I tried copying the Yoshi's Protractor task settings for the `webdriver-manager update`:
```
webdriver-manager update --versions.chrome=2.46 --gecko=false --standalone=true 
```
( I used a fixed chrome version, temporarily)

and it is failing in Travis CI, for a :
```
E/launcher - unknown error: Chrome failed to start: exited abnormally
@unidriver/protractor:   (unknown error: DevToolsActivePort file doesn't exist)
```

This is [the build](https://travis-ci.org/wix-incubator/unidriver/builds/516766532)

#### Without Params
I tried also [without any config](https://travis-ci.org/wix-incubator/unidriver/builds/516767159#L526)
```
webdriver-manager update && protractor ./dist/protractor.conf.js
```
Same error.

#### Might be a Related Issue

This Selenium issue looks related, but I couldn't understand what is the solution there.
https://github.com/SeleniumHQ/selenium/issues/6077

And why is it working for Yoshi in TeamCity CI ?

Any help would be appreciated.
Thanks.

**Comment by GabiGrin** [comment]
@hepiyellow I was able to fix it quickly by searching the error, finding this [repo](https://github.com/kendaleiv/angular-testing/blob/master/protractor.conf.js) and copying stuff from there

**Pull request merged**